### DISCUSSION TITLE
Foldable "closed" Property

### DISCUSSION THREAD

**Discussion by ArianYN** [sqlpage/SQLPage#594] [Ideas]
It would be nice to have a property for the new Foldable Component, where you can do like:
`
TRUE as closed,
`
If you want the Foldable to start off closed.

Then also if we can add an icon at the front of the foldable would be nice

Also a little extra, when using the Foldable Component under some other Component, the space between those 2 components is very tiny. It would be nice if there was a bit more space added between that. (This is especially noticable when the foldeable is under a title component)

I also just found out that when you have more "Foldables" on 1 page, opening or closing a single one, opens and closes ALL of them. I think they are connected somehow...

**Comment by lovasoa** [comment]
I updated the component, added a new `expanded` property and a bottom margin, as well as the ability to set custom classes (to personalize margins and padding if necessary).

**Reply by ArianYN** [reply]
That's great. Only one thing, ALL the foldables on a page, expand whenever I click 1 of them.

**Reply by lovasoa** [reply]
Can you share a minimal reproduction? Do you see that happening with the example from the documentation too?

**Reply by ArianYN** [reply]
I am trying this

Keep in mind i have tried this with "class" aswell as use both "id" and "class" in the top-level also. None of them work. They all still expand

```sql
SELECT
    'foldable' as component;
SELECT
    'foldable1' as id,
    'Foldable Test 1' as title,
    'This is the first foldable test' as description_md;

SELECT
    'foldable' as component;
SELECT
    'foldable2' as id,
    'Foldable Test 2' as title,
    'This is the second foldable test' as description_md;
 ```

**Reply by lovasoa** [reply]
Indeed, we have a bug when we have multiple foldables on the same page !

**Reply by lovasoa** [reply]
@ArianYN , I fixed it.

Thanks again for reporting the issue

**Reply by ArianYN** [reply]
Amazing :) Thanks.
This is available in the newest pre-release on docker hub right?
